include/roboteam_ai/world/BallPossession.h

```diff
@@ -11,33 +7,89 @@
 namespace rtt {
 namespace ai {
 
+/**
+ * Computes and stores the ball possession state which indicates which team controls the ball (can also be both/no team).
+ * @author Created by: Rolf van der Hulst <br>
+ *         Documented by: Haico Dorenbos
+ * @since 2019-04-15
+ */
 class BallPossession {
     FRIEND_TEST(BallPossessionTest, team_far_or_close_to_ball);
     FRIEND_TEST(BallPossessionTest, it_properly_computes);
+
+private:
+    const double MIDDLE_LINE_X = 0.0; // The x coordinate that corresponds to the middle of the field.
```

Comment:
I think this goes a little far

Reply:
The middle line position is a magic value, we could for example also decide that our left line is at x = 0.0. I indeed agree that we still have some kind of code cloning in this case, but it is much better than in the previous case.